code/modules/xenoarchaeology/traits/xenoartifact_minors.dm

```diff
@@ -0,0 +1,430 @@
+	range = Z.max_range+3
+
+/obj/effect/proc_holder/spell/targeted/xeno_senitent_action/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	for(var/atom/M in targets)
+		if(xeno)
+			xeno.true_target = list(M)
+			xeno.default_activate(xeno.charge_req+50)
+			charge_max = xeno.cooldown+xeno.cooldownmod
+
+/datum/xenoartifact_trait/minor/sentient/Destroy(force, ...)
+	. = ..()
+	QDEL_NULL(man) //Kill the inner person. Otherwise invisible runs around
+	QDEL_NULL(S)
+
+/obj/effect/mob_spawn/sentient_artifact
+	death = FALSE
+	name = "Sentient Xenoartifact"
+	short_desc = "You're a maleviolent sentience, possesing an ancient alien artifact."
+	flavour_text = "Return to your master..."
+	use_cooldown = TRUE
+	invisibility = 101
+	var/obj/item/xenoartifact/artifact
+
+/obj/effect/mob_spawn/sentient_artifact/Initialize(mapload, var/obj/item/xenoartifact/Z)
+	if(!Z)
+		qdel(src)
+		return FALSE
+	artifact = Z
+	return ..()
+
+/obj/effect/mob_spawn/sentient_artifact/create(ckey, name)
+	var/datum/xenoartifact_trait/minor/sentient/S = artifact.get_trait(/datum/xenoartifact_trait/minor/sentient)
+	S.setup_sentience(artifact, ckey)
+
+//============
+// Delicate, makes the artifact have limited uses
+//============
+/datum/xenoartifact_trait/minor/delicate
+	desc = "Fragile"
+	label_desc = "Fragile: The Artifact is poorly made. Continuous use will destroy it."
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/delicate/on_init(obj/item/xenoartifact/X)
+	X.max_integrity = pick(200, 300, 500, 800, 1000)
+	X.obj_integrity = X.max_integrity
+	X.alpha = X.alpha * 0.55
+
+/datum/xenoartifact_trait/minor/delicate/activate(obj/item/xenoartifact/X, atom/user)
+	if(X.obj_integrity)
+		X.obj_integrity -= 100
+	else if(X.obj_integrity <= 0)
+		X.visible_message("<span class='danger'>The [X.name] shatters!</span>", "<span class='danger'>The [X.name] shatters!</span>")
+		var/obj/effect/decal/cleanable/ash/A = new(get_turf(X))
+		A.color = X.material
+		playsound(get_turf(X), 'sound/effects/glassbr1.ogg', 50, TRUE) 
+		qdel(X)
+
+//============
+// Aura, adds everything in the vicinity to the target list
+//============
+/datum/xenoartifact_trait/minor/aura
+	desc = "Expansive"
+	label_desc = "Expansive: The Artifact's surface reaches towards every creature in the room. Even the empty space behind you..."
+	blacklist_traits = list(/datum/xenoartifact_trait/major/timestop, /datum/xenoartifact_trait/minor/long)
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/aura/on_init(obj/item/xenoartifact/X)
+	X.max_range += 2
+
+/datum/xenoartifact_trait/minor/aura/activate(obj/item/xenoartifact/X)
+	for(var/atom/M in oview(min(X.max_range, 5), get_turf(X.loc)))
+		if(X.true_target.len >= XENOA_MAX_TARGETS)
+			return
+		var/obj/item/I = M
+		if(istype(M, /mob/living))
+			X.true_target |= X.process_target(M)
+		else if(istype(I) && !(I.anchored))
+			X.true_target |= X.process_target(I)
```

Comment:
How do you actually figure this out in game without knowing the artifact range from code diving?